### PR TITLE
Fixed Multicursor backspacing

### LINKED ISSUE #7205: Multicursor backspacing doesn't properly handle tabs
We have special logic to delete to the previous tab stop, but that logic doesn't work with multiselect.

To see this in action, use multiselect on the function below before the print statements and try to delete both. Only one of the tabs will be deleted and the other remains as is.
```
def foo():
    print('bar1')
    print('bar2')
```

https://github.com/jupyter/notebook/pull/4880 fixes this for jupyter notebooks

### PR DESCRIPTION
**Pull request by bthayer2365**
## References

fixes #7205

Other references:
https://github.com/jupyter/notebook/pull/4880
https://github.com/jupyter/notebook/issues/4796
## Code changes

Iterates through selections, and handle backspacing appropriately. There are three cases: 
 1. Selection
    - Delete everything in selection
 2. Leading spaces
    - Delete to previous indentation
 3. Other
    - Delete previous character

3\. Can be broken down into the cases of the beginning of the line too, but they behave as you would expect

This solution is similar but not exactly the same as for the same problem in jupyter notebook https://github.com/jupyter/notebook/pull/4880

## User-facing changes

Basically, backspacing when using multicursor works as expected now whereas before it didn't handle tabs correctly.

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by blink1073** [comment]
Thanks @bthayer2365!  I don't have a working laptop at the moment to test but this logic looks great to me!

**Comment by bthayer2365** [comment]
Great, looking forward to you testing it!

**Comment by blink1073** [comment]
Very nice!  This is identical to the behavior I see in VS Code.  I think these build errors were temporary, kicking the builds.

**Comment by bthayer2365** [comment]
Linux integrity errors:
```
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1364:5 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1365:5 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1366:10 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1368:7 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1369:7 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1370:7 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1374:9 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1377:11 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1378:11 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1382:23 - == should be ===
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1383:27 - != should be !==
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1384:15 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1384:19 - Duplicate variable: 'from'
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1391:13 - Forbidden 'var' keyword, use 'let' or 'const' instead
ERROR: /home/vsts/work/1/s/packages/codemirror/src/editor.ts:1391:17 - Duplicate variable: 'from'
```
I fixed all the var/let issues. Changing var to let for `from` should remove the duplicate variable error. Also fixed `==` and `!=` errors.

Windows JS error:
```
2019-10-24T04:11:02.3193805Z @jupyterlab/test-codemirror: Failed to collect coverage from D:\a\1\s\packages\codemirror\src\editor.ts
2019-10-24T04:11:02.3195636Z @jupyterlab/test-codemirror: ERROR: ..\..\packages\codemirror\src\editor.ts: Emit skipped
2019-10-24T04:11:02.3196024Z @jupyterlab/test-codemirror: STACK: TypeError: ..\..\packages\codemirror\src\editor.ts: Emit skipped
```
I don't know how to interpret these. Maybe changing things from var to let will clear that up, but if not, could you let me know what to do?

The Linux Docs error doesn't seem to be my fault, so I'm leaving it as is.

**Comment by blink1073** [comment]
The error you saw is being tracked in https://github.com/jupyterlab/jupyterlab/issues/6415.   Yes, the docs failure is temporary until we release 1.2.  Thanks again, and congratulations on your first contribution to JupyterLab!

**Comment by jasongrout** [comment]
@blink1073 - should this be backported to 1.2?

**Comment by blink1073** [comment]
@meeseeksdev backport to 1.x
